creator-node/src/utils/registeredNodes.ts

```diff
@@ -0,0 +1,65 @@
+import type Logger from 'bunyan'
+import { Redis } from 'ioredis'
+
+const THIRTY_MINUTES_IN_SECONDS = 60 * 30
+
+type Service = {
+  owner: string
+  endpoint: string
+  spID: number
+  type: 'discovery-node' | 'content-node'
+  blockNumber: number
+  delegateOwnerWallet: string
+}
+
+/**
+ * Get all Discovery Nodes registered on chain
+ * Fetches from Redis if available, else fetches from chain and updates Redis value
+ * @returns {Promise<Service[]>} array of SP objects with schema { owner, endpoint, spID, type, blockNumber, delegateOwnerWallet }
+ */
+export const getAllRegisteredDNodes = async ({
+  libs,
+  logger,
+  redis
+}: {
+  libs: any
+  logger: Logger
+  redis: Redis
+}): Promise<Service[]> => {
+  const cacheKey = 'all_registered_dnodes'
+
+  let DNodes
+  try {
```

Comment:
concerned about error rates if this ever fails, maybe good to add retries (`utils/asyncRetry`)

Reply:
want to minimize accidentally 403 responses

Reply:
can add the retries, but not sure how we'd get 403 trying to fetch the registered discovery nodes

Reply:
ah gotchu. yup makes sense. i added the retries :)